Watchman.Discord/Areas/Administration/Controllers/AdministrationController.cs

```diff
@@ -93,28 +93,54 @@ public async Task SetRoleAsSafe(DiscordRequest request, Contexts contexts)
                 throw new NotEnoughArgumentsException();
             }
 
-            var roleName = args[0].Value;
-            var toSetAsSafe = args[1].Value == "safe";
+            if (args.GroupBy(x => x.Value)
+                .Select(x => x.First()).Count() != args.Length)
+            {
+                throw new ArgsAreDuplicatedException();
+            }
+
+            var lastArg = args[args.Length - 1].Value;
 
-            var serverRole = _usersRolesService.GetRoleByName(roleName, contexts.Server);
-            if (serverRole == null)
+            if (lastArg.ToLower() != "safe"
+                && lastArg.ToLower() != "unsafe")
             {
-                throw new RoleNotFoundException(roleName);
+                throw new NotEnoughArgumentsException();
             }
 
-            if (toSetAsSafe)
+            for (int i = 0; i < args.Length - 1; i++)
             {
-                var command = new SetRoleAsSafeCommand(roleName, contexts.Server.Id);
-                await _commandBus.ExecuteAsync(command);
+                var roleName = args[i].Value;
+                var serverRole = _usersRolesService.GetRoleByName(roleName, contexts.Server);
+
+                if (serverRole == null)
+                {
+                    throw new RoleNotFoundException(roleName);
+                }
             }
-            else
+
+            for (int i = 0; i < args.Length - 1; i++)
```

Comment:
Jak masz tak duże metody kontrolera, to pomyśl czy dałoby się jakoś wrzucić to do serwisu 
(nie zawsze to jest możliwe, ale tu masz sporo pętli itp, może je dałoby się tam upchnąć)